src/otx/algorithms/common/adapters/torch/dataloaders/samplers/otx_sampler.py

```diff
@@ -0,0 +1,99 @@
+    since DataLoader reinitialized at every epoch. So, in the small dataset case,
+    OTXSampler repeats the dataset to enlarge the iterations per epoch.
+
+    In the large dataset, the useful information is not totally linear relationship with the number of datasets.
+    It is close to the log scale relationship, rather.
+
+    So, this sampler samples or repeats the datasets acoording to the statistics of dataset.
+
+    Args:
+        dataset (Dataset): A built-up dataset
+        samples_per_gpu (int): batch size of Sampling
+        use_adaptive_repeats (bool): Flag about using adaptive repeats
+        num_replicas (int, optional): Number of processes participating in
+            distributed training. By default, :attr:`world_size` is retrieved from the
+            current distributed group.
+        rank (int, optional): Rank of the current process within :attr:`num_replicas`.
+            By default, :attr:`rank` is retrieved from the current distributed
+            group.
+        shuffle (bool, optional): Flag about shuffling
+        coef (int, optional): controls the repeat value
+        min_repeat (float, optional): minimum value of the repeat dataset
+    """
+
+    def __init__(
+        self,
+        dataset: Dataset,
+        samples_per_gpu: int,
+        use_adaptive_repeats: bool,
+        num_replicas: int = 1,
+        rank: int = 0,
+        shuffle: bool = True,
+        coef: float = -0.7,
+        min_repeat: float = 1.0,
+    ):
+
+        self.dataset, _ = unwrap_dataset(dataset)
+        self.samples_per_gpu = samples_per_gpu
+        self.num_replicas = num_replicas
+        self.rank = rank
+        self.shuffle = shuffle
+        self.repeat = self._get_proper_repeats(use_adaptive_repeats, coef, min_repeat)
+
+        self.num_samples = math.ceil(len(self.dataset) * self.repeat / self.num_replicas)
+        self.total_size = self.num_samples * self.num_replicas
+
+    def _get_proper_repeats(self, use_adaptive_repeats: bool, coef: float, min_repeat: float):
+        """Calculate the proper repeats with considering the number of iterations."""
+        n_repeats = 1
+        if use_adaptive_repeats:
+            # NOTE
+            # Currently, only support the integer type repeats.
+            # Will support the floating point repeats and large dataset cases.
+            n_iters_per_epoch = math.ceil(len(self.dataset) / self.samples_per_gpu)
+            n_repeats = math.floor(max(coef * math.sqrt(n_iters_per_epoch - 1) + 5, min_repeat))
+            logger.info("OTX Sampler: adaptive repeats enabled")
+            logger.info(f"OTX will use {n_repeats} times larger dataset made by repeated sampling")
+        return n_repeats
+
+    def __iter__(self):
+        """Iter."""
+        indices = list(range(len(self.dataset)))
+
+        # produce repeats e.g. [0, 0, 0, 1, 1, 1, 2, 2, 2....]
+        indices = [x for x in indices for _ in range(self.repeat)]
```

Comment:
Would this be better in performance? I'm not sure.
indices = indices*self.repeat
or
indices = np.repeat(indices, self.repeat)

Reply:
Hmm, `np.repeat` seems good. I just used the pre-existed one. https://github.com/open-mmlab/mmpretrain/blob/732b0f4c9869d952eca7d66cb82ad0be8d83cc10/mmpretrain/datasets/samplers/repeat_aug.py#L74